### DISCUSSION TITLE
Include command files with developer module

### DISCUSSION THREAD

**Discussion by avazon** [oqtane/oqtane.framework#3913] [Q&A]
I run Oqtane in Azure and have a developer module running an oqtane scheduled job. I would like to include an extra console application in a subdirectory for maintance and debug of the module (from a shell or command prompt).
I look in the nuspec file of the module and in section "files" files are defined with source and destination. Target is "lib\net8.0". But I would like to avoid having the files in the same directory as Oqtane and prefer a subdirectory. Does anyone have a suggestion howto?

**Comment by sbwalker** [comment]
@avazon it is not clear from your description what do you mean by "an extra console application in a subdirectory for maintance and debug of the module"? I am guessing that you want to be able to trigger the Oqtane scheduled job so that you can test and debug it? 

There are a few ways I have accomplished this in the past:

1. You should use the approach included in the default Module Template where the Oqtane.Server project is linked to your module solution  as a startup project... which means you launch Oqtane from your module solution within Visual Studio... and you can place a breakpoint on the ExecuteJob() method in your module code and step through the logic line by line.

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/67f38457-5270-44b0-9aeb-68a3940029bb)

2. When you are running Oqtane you can edit the job in the Schedule Jobs UI and set its Next Execution property to a date in the past. Once you save the job, it will execute within 1 minute (as long as the application is running).

3.  You can architect your module to include a Manager class. Migrate the logic from your scheduled job to this Manager class so that the job itself simply calls the Manager class. And then you can add a Controller which contains an API method that can invoke this same Manager class. Then you would have a way to trigger the logic from a Url (or you could create a button in your Blazor UI to invoke it).

I do not believe there is a need for a console app if you follow the approaches above.
